user-operator/src/main/java/io/strimzi/operator/user/UserOperatorConfig.java

```diff
@@ -46,512 +58,381 @@ public class UserOperatorConfig {
+        Map<String, Object> generatedMap = ConfigParameter.define(map, CONFIG_VALUES);
 
-        String caNamespace = map.get(UserOperatorConfig.STRIMZI_CA_NAMESPACE);
-        if (caNamespace == null || caNamespace.isEmpty()) {
-            caNamespace = namespace;
+        if (generatedMap.get(STRIMZI_CA_NAMESPACE) == null || String.valueOf(generatedMap.get(STRIMZI_CA_NAMESPACE)).isEmpty()) {
+            generatedMap.put(STRIMZI_CA_NAMESPACE, generatedMap.get(NAMESPACE.key()));
         }
 
-        String secretPrefix = map.get(UserOperatorConfig.STRIMZI_SECRET_PREFIX);
-        if (secretPrefix == null || secretPrefix.isEmpty()) {
-            secretPrefix = DEFAULT_SECRET_PREFIX;
-        }
-
-        List<String> maintenanceWindows = parseMaintenanceTimeWindows(map.get(UserOperatorConfig.STRIMZI_MAINTENANCE_TIME_WINDOWS));
-
-        Properties kafkaAdminClientConfiguration = parseKafkaAdminClientConfiguration(map.get(UserOperatorConfig.STRIMZI_KAFKA_ADMIN_CLIENT_CONFIGURATION));
-
-        return new UserOperatorConfig(namespace, reconciliationInterval, kafkaBootstrapServers, labels,
-                caCertSecretName, caKeySecretName, clusterCaCertSecretName, euoKeySecretName, caNamespace, secretPrefix,
-                aclsAdminApiSupported, kraftEnabled, clientsCaValidityDays, clientsCaRenewalDays,
-                scramPasswordLength, maintenanceWindows, kafkaAdminClientConfiguration, operationTimeout, workQueueSize,
-                controllerThreadPoolSize, cacheRefresh, batchQueueSize, batchMaxBlockSize, batchMaxBlockTime,
-                userOperationsThreadPoolSize);
+        return new UserOperatorConfig(generatedMap);
     }
 
     /**
-     * Parse the Kafka Admin Client configuration from the environment variable
-     *
-     * @param configuration The configuration from the environment variable. Null if no configuration is set.
-     *
-     * @return  The properties object with the configuration
+     * @return Set of configuration key/names
      */
-    /* test */ static Properties parseKafkaAdminClientConfiguration(String configuration) {
-        Properties kafkaAdminClientConfiguration = new Properties();
-
-        if (configuration != null)   {
-            try {
-                kafkaAdminClientConfiguration.load(new StringReader(configuration));
-            } catch (IOException | IllegalArgumentException e)   {
-                throw new InvalidConfigurationException("Failed to parse " + UserOperatorConfig.STRIMZI_KAFKA_ADMIN_CLIENT_CONFIGURATION + " configuration", e);
-            }
-        }
-
-        return kafkaAdminClientConfiguration;
+    public static Set<String> keyNames() {
+        return Collections.unmodifiableSet(CONFIG_VALUES.keySet());
     }
 
     /**
-     * @return  Clients CA validity in days
+     * Gets the configuration value corresponding to the key
+     * @param <T> type of value
+     * @param value instance of Value class
+     * @return configuration value w.r.t to the key
```

Comment:
Again, formatting the JavaDocs nicely as it was in the old code can do magic in how the Javadoc is readable. So does better capitalization.

```suggestion
     * Gets the configuration value corresponding to the key
     * 
     * @param <T>     Type of value
     * @param value   Instance of Value class
     * 
     * @return   Configuration value w.r.t to the key
```